### ISSUE TITLE
Dependently-typed core language & low-level representation

### ISSUE THREAD

**Issue by cwgoes** [cryptiumlabs/juvix#2]
To some degree, must be chosen in parallel.

Either dependently-typed core or dependently-typed core plus un-typed IR, and necessarily a low-level evaluation target.

Core language candidates / inspirations:
- Idris Core
- Blodwen Core
- Cedille Core
- Formality (several variations)

We don't need to reinvent the wheel here; the key part is evaluation efficiency (and simplicity).

Evaluation targets:
- Interaction nets (interaction combinators only)
- Interaction nets (multiple node types)
- WASM / Michelson / EVM (directly, call-by-value)

**Comment by cwgoes** [comment]
Closing in favor of more up-to-date issues.
